Tests/SourceKitLSPTests/SemanticTokensTests.swift

```diff
@@ -0,0 +1,252 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import LanguageServerProtocol
+import LSPTestSupport
+import SKTestSupport
+import SourceKitLSP
+import XCTest
+
+final class SemanticTokensTests: XCTestCase {
```

Comment:
Do you think it would be useful to have some kind of debugging option with which we can turn lexical/syntactic/semantic tokens on or off as a source for the semantic tokens request? That way, we could add tests that selectively only enable one of them and verify that they are actually providing information which is incorporated in the response.

Or do you think that would be too difficult to wire up?

Reply:
I think that would definitely be nice to have, ideally we could even expose that as an LSP config options so clients could selectively enable/disable different 'kinds' of tokens (lexical/syntactic/semantic). Perhaps it would be a good candidate for a future PR, since this one has already gotten pretty large?

Reply:
Sounds good to me 👍.